### PR TITLE
FuriHal: interrupt priorities and documentation

### PR DESCRIPTION
**Pull request by skotopes**
# What's new

- FuriHal: interrupt priorities and documentation

# Verification 

- Full set of tests

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: targets/f7/furi_hal/furi_hal_interrupt.h [hunk 1]
```diff
@@ -53,27 +53,46 @@ typedef enum {
...
+
 /** Initialize interrupt subsystem */
 void furi_hal_interrupt_init();
 
-/** Set ISR and enable interrupt with default priority
- * We don't clear interrupt flags for you, do it by your self.
- * @param index - interrupt ID
- * @param isr - your interrupt service routine or use NULL to clear
- * @param context - isr context
+/** Set ISR and enable interrupt with default priority We don't clear interrupt
```
Comment (gsurkov) [inline targets/f7/furi_hal/furi_hal_interrupt.h]: Suggesting more neutral wording:
"Interrupt flags are not cleared automatically, the user code must do it by calling ..."

File: targets/f7/furi_hal/furi_hal_interrupt.h [hunk 2]
```diff
@@ -53,27 +53,46 @@ typedef enum {
...
+    FuriHalInterruptPriorityNormal =
+        0, /**< Normal(default) priority level, you can use ISR-safe OS primitives */
+    FuriHalInterruptPriorityHigh =
+        1, /**< Higher than normal priority level, you can use ISR-safe OS primitives */
+    FuriHalInterruptPriorityVeryHigh =
+        2, /**< Highest priority level, you can use ISR-safe OS primitives */
+
+    /* Special group, read docs first(ALL OF THEM: especially FreeRTOS configLIBRARY_MAX_SYSCALL_INTERRUPT_PRIORITY) */
+    FuriHalInterruptPriorityKamiSama =
+        3, /**< Forget about thread safety, you are god now, no one can prevent you from fucking up OS critical section, you are not allowed to use any OS primitives, but who can stop you? Use this priority only for direct hardware interaction with LL HAL. */
```
Comment (gsurkov) [inline targets/f7/furi_hal/furi_hal_interrupt.h]: Swears? Better provide a short explanation why it is dangerous.

Comment (skotopes) [reply targets/f7/furi_hal/furi_hal_interrupt.h]: fixed

File: targets/f7/furi_hal/furi_hal_interrupt.h (hunk 1 above)
Comment (skotopes) [reply targets/f7/furi_hal/furi_hal_interrupt.h]: done

File: targets/f7/furi_hal/furi_hal_interrupt.c [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 #define TAG "FuriHalInterrupt"
 
-#define FURI_HAL_INTERRUPT_DEFAULT_PRIORITY (configLIBRARY_MAX_SYSCALL_INTERRUPT_PRIORITY)
+#define FURI_HAL_INTERRUPT_DEFAULT_PRIORITY (configLIBRARY_MAX_SYSCALL_INTERRUPT_PRIORITY + 2)
```
Comment (DrZlo13) [inline targets/f7/furi_hal/furi_hal_interrupt.c]: +5 plz

**Review by hedger — APPROVED** [review]

**Review by hedger — APPROVED** [review]

**Pull request merged**